lib/yargs.ts

```diff
@@ -0,0 +1,1101 @@
+import { tokenizeArgString } from './tokenize-arg-string.js'
+import type {
+  ArgsInput,
+  Arguments,
+  ArrayFlagsKey,
+  ArrayOption,
+  CoerceCallback,
+  Configuration,
+  DefaultValuesForType,
+  DefaultValuesForTypeKey,
+  DetailedArguments,
+  Flag,
+  Flags,
+  FlagsKey,
+  StringFlag,
+  BooleanFlag,
+  NumberFlag,
+  ConfigsFlag,
+  CoercionsFlag,
+  Options,
+  OptionsDefault,
+  YargsParserMixin
+} from './yargs-parser-types.js'
+import type { Dictionary, ValueOf } from './common-types.js'
```

Comment:
Aren't these files `.ts`? I'm confused by the explicit `.js` extensions.

Reply:
this is an oddity I learned during this process, TypeScript doesn't allow you to include `.ts` extensions, they need to instead reference the path that will eventually be the target extension.  So, to target code that works for ESM and Deno, you need to include the `.js` extension, which will then be included in the compiled code.

Reply:
Wow, TIL.